libcontainer/setns_init_linux.go

```diff
@@ -34,10 +34,12 @@ func (l *linuxSetnsInit) getSessionRingName() string {
 
 func (l *linuxSetnsInit) Init() error {
 	if !l.config.Config.NoNewKeyring {
-		if err := selinux.SetKeyLabel(l.config.ProcessLabel); err != nil {
-			return err
+		if l.config.ProcessLabel != "" {
+			if err := selinux.SetKeyLabel(l.config.ProcessLabel); err != nil {
+				return err
+			}
+			defer selinux.SetKeyLabel("") //nolint: errcheck
 		}
-		defer selinux.SetKeyLabel("") //nolint: errcheck
```

Comment:
Do we know what the intent was for these empty `selinux.SetKeyLabel("")` ? See my comment on https://github.com/opencontainers/selinux/pull/214#issuecomment-2255435492

Reply:
I think it's used to clear the label for the next keyring created by the current process once we got an error when we start/exec the container. Because this label can't be inherited by sub-process, so the initial value should be "", so we can set it to an empty string directly to clear it.

Reply:
can we use unix.KeyctlString() set  keylabel ? @lifubang

Reply:
> can we use unix.KeyctlString() set keylabel ?

I think no.

Reply:
> Do we know what the intent was for these empty `selinux.SetKeyLabel("")` ? See my comment on [opencontainers/selinux#214 (comment)](https://github.com/opencontainers/selinux/pull/214#issuecomment-2255435492)

Some are added by commit cd96170c, and some are by aa3fee6c, which provides an explanation:

> should also call SetProcessLabel("") after the container starts
> to go back to the default SELinux behaviour.

Given the fact that Init() never returns unless there's an error (because it ends with exec), those defers are only called in case of an error.